src/lab12.py

```diff
@@ -447,8 +446,7 @@ def run(self):
             needs_quit = self.needs_quit
             self.lock.release()
             if needs_quit:
-                self.handle_quit()
-                return
+                return self.handle_quit()
```

Comment:
Is this just a clever trick to avoid two lines?

Reply:
You're right, it's dumb.